src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/DateTimeValueResolver.php

```diff
@@ -53,15 +59,20 @@ public function resolve(Request $request, ArgumentMetadata $argument): array
         }
 
         $class = \DateTimeInterface::class === $argument->getType() ? \DateTimeImmutable::class : $argument->getType();
+
+        if ($this->clock && !$value) {
+            $date = $this->clock->withTimeZone(date_default_timezone_get())->now();
+
+            return $date instanceof $class ? [$date] : [$class::createFromInterface($date)];
```

Comment:
we probably this same conversion in the `if ($value instanceof \DateTimeInterface) {` above to make sure we always return `$class`

Reply:
I updated the code to apply conversion to already provided value.

Reply:
thinking about that, this conversion should probably be applied in a separate PR as this current PR will wait for 6.3 while that change is a bugfix

Reply:
Done and rebased on https://github.com/symfony/symfony/pull/48346